src/api/course/index.ts

```diff
@@ -84,23 +94,137 @@ export class CourseApi {
       if (storeData) {
         this.emitCourseInfo(true)
       }
+      // register delta handler
+      ;(this.app as any).registerDeltaInputHandler(this.deltaHandler)
       resolve()
     })
   }
 
+  /** handle and process incoming course stream data */
+  private deltaHandler = (
+    delta: ValuesDelta,
+    next: (delta: ValuesDelta) => void
+  ) => {
+    this.processStreamSources(delta)
+    next(delta)
+  }
+
+  /** Process stream <destination>.nextPoint data
+   * Note: Stream source cannot override destination set by API!
+   * Destination is set when:
+   * 1. There is no current destination
+   * 2. msg source matches current Destination source
+   * 3. Destination Position is changed.
+   */
+  private async processStreamSources(delta: ValuesDelta) {
```

Comment:
I would name this something like `processV1DestinationDeltas` or something like that - StreamSources is really ambiguous.

Reply:
Updated function name.